File: r/difference-of-squares/difference-of-squares.R
# this is a stub function that takes a natural_number
# and should return the difference-of-squares as described
# in the README.md
difference_of_squares <- function(natural_number) {
  square_of_sum <- (natural_number / 2 * (natural_number + 1)) ** 2
  sum_of_squares <- (natural_number * (natural_number + 1) * (2 * natural_number + 1)) / 6

  square_of_sum - sum_of_squares
}

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const num_float: f32 = @floatFromInt(number);
    const sum: usize = @intFromFloat(num_float / 2.0 * (num_float + 1.0));
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    return number * (number + 1) * (2 * number + 1) / 6;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
